### PR TITLE
Defer scalar API calls until they are needed

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/5846

In practice this means not calling scalar until one of the following things happens:
* The user opens the integration manager
* The user starts a conference call
* The user has a widget open

We no longer eagerly check for scalar auth everywhere, which was causing multiple calls to various endpoints at useless times. This means that the manage integrations button doesn't report errors on itself anymore (but the integration manager dialog does), the sticker picker doesn't check auth until needed, and the app drawer doesn't check for auth (instead letting the widgets do that).

After this PR, we make zero calls to scalar when switching rooms (unless the destination room has a widget) and we make zero calls on startup (unless the landing room has a widget). 

The behaviour of having an empty (not undefined) `integrations_ui_url` in the config still works - the integrations server is considered disabled (see screenshots below).

Tested with both Modular (Scalar) and Dimension - both are happy with this change as far as I can tell.

Integrations enabled:
image

Integrations disabled:
image
image

New integration manager dialog states:
image
image

### CODE REVIEW

File: src/components/views/rooms/Stickerpicker.js [hunk 1]
```diff
@@ -63,14 +66,34 @@ export default class Stickerpicker extends React.Component {
         };
     }
 
-    _removeStickerpickerWidgets() {
+    _acquireScalarClient() {
+        if (this.scalarClient) return Promise.resolve(this.scalarClient);
+        if (SdkConfig.get().integrations_ui_url && SdkConfig.get().integrations_rest_url) {
```
Comment (bwindels) [inline src/components/views/rooms/Stickerpicker.js]: can we use `ScalarAuthClient.isPossible()` here as well?

File: src/IntegrationManager.js [hunk 2]
```diff
@@ -15,64 +16,25 @@ limitations under the License.
...
-        !(global.mxIntegrationManager.client && global.mxIntegrationManager.client.hasCredentials())) {
-      console.error("Scalar error", global.mxIntegrationManager);
-      return;
+    /**
+     * Launch the integrations manager on the specified integration page
+     * @param  {string} integName integration / widget type
+     * @param  {string} integId   integration / widget ID
+     * @param  {function} onFinished Callback to invoke on integration manager close
+     */
+    static async open(integName, integId, onFinished) {
```
Comment (bwindels) [inline src/IntegrationManager.js]: nit: As this is only called from `FromWidgetPostMessageApi` on a `integration_manager_open` message, should we just move this code there?

**Comment by turt2live** [comment]
`onStarterLinkClick` is unaffected by this change, and doesn't spam calls when it doesn't need to.

**Comment by bwindels** [comment]
Got a lint error though:
```
/src/components/views/rooms/Stickerpicker.js 22:8 error 'SdkConfig' is defined but never used no-unused-vars
```

**Pull request merged**